docs/csharp/language-reference/operators/patterns.md

```diff
@@ -41,6 +41,37 @@ Beginning with C# 9.0, for that purpose you can use the *type pattern* as the fo
 
 Like a declaration pattern, a type pattern matches an expression when an expression result is non-null and its runtime type satisfies any of the conditions listed above.
 
+## Constant pattern
+
+Beginning with C# 7.0, you use the *constant pattern* to test if an expression result equals a specified constant, as the following example shows:
+
+:::code language="csharp" source="snippets/patterns/ConstantPattern.cs" id="BasicExample":::
+
+In a constant pattern, you can use any constant expression, such as:
+
+- an [integer](../builtin-types/integral-numeric-types.md) or [floating-point](../builtin-types/floating-point-numeric-types.md) numerical literal
+- a [char](../builtin-types/char.md) or a [string](../builtin-types/reference-types.md#the-string-type) literal
+- a Boolean value `true` or `false`
+- an [enum](../builtin-types/enum.md) value
+- the name of a declared [const](../keywords/const.md) field or local
+- `null`
+
+Use the constant pattern to check for `null`, as the following example shows:
+
+:::code language="csharp" source="snippets/patterns/ConstantPattern.cs" id="NullCheck":::
+
+The compiler guarantees that no user-overloaded equality operator `==` is invoked when expression `x is null` is evaluated.
+
+## Discard pattern
+
+Beginning with C# 8.0, you use the *discard pattern* `_` to match any expression, including `null`, as the following example shows:
+
+:::code language="csharp" source="snippets/patterns/DiscardPattern.cs" id="BasicExample":::
+
+The discard pattern is useful in `switch` expressions. If none of a `switch` expression's patterns matches an input, the runtime throws an exception. To guarantee that a `switch` expression handles all possible values of its input as you want, provide an arm with the discard pattern. As a `switch` expression evaluates its arms in text order, the arm with the discard pattern must be last.
```

Comment:
It might be useful to mention that the compiler emits a warning if the arms of a `switch` expression don't handle all possible inputs.

This one may be surprising for some readers when the switch variable is an `enum`.

Reply:
@BillWagner I think that mention should be not in this article, but in the `switch` expression article; that article even has the [dedicated section](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression#non-exhaustive-switch-expressions) for that. What do you think?

> This one may be surprising for some readers when the switch variable is an `enum`.

Yes, I've tested that scenario :). I think that the compiler message is very good and clear. For example, switching on `DayOfWeek` enum:

> Warning CS8524 The switch expression does not handle some values of its input type (it is not exhaustive) involving an unnamed enum value. For example, the pattern '(System.DayOfWeek)7' is not covered.

I would say such a warning is self-explanatory.

Reply:
I can update the switch expression article in the main branch accordingly